src/Nethermind/Nethermind.HashLib/IHash.cs

```diff
@@ -39,6 +40,7 @@ public interface IHash
         HashResult ComputeString(string a_data);
         HashResult ComputeString(string a_data, Encoding a_encoding);
         HashResult ComputeBytes(byte[] a_data);
+        HashResult ComputeBytes(ReadOnlySpan<byte> a_data);
```

Comment:
It may still be safer / clearer to have: void ComputeBytes(ReadOnlySpan<byte> input, Span<byte> output), so the caller knows that they own the output.

byte[] output = new byte[32];
Hash.ComputeBytes(input, output);
Bytes32 bytes32 = Bytes32.CreateWith(output);
return bytes32;

Or have a pattern similar to the String.Create with no allocation.

Reply:
String.Create won't work, as can't pass ReadOnlySpan<byte> as generic to create function.

Reply:
I started writing it and started falling asleep in the middle and it ended a mess that I had to revert. But it should be easy to add, just decided it was not a priority. Also allocating HashResult is additional overhead. So yeah - agreed with you but we achieved 80% improvement already.